sectionproperties/pre/sections.py

```diff
@@ -2002,129 +2544,115 @@ class PolygonSection(Geometry):
-        if r_in > a_in:
-            r_in = a_in
-            circle = True
-        else:
-            circle = False
+    # calculate external radius, if r_in is zero, r_out also is zero
+    if r_in == 0:
+        r_out = 0
+        n_r = 1
+    else:
+        r_out = r_in + t
 
-        # calculate external radius, if r_in is zero, r_out also is zero
-        if r_in == 0:
-            r_out = 0
-            n_r = 1
-        else:
-            r_out = r_in + t
+    # equivalent side length of half the corner radii triangular segment
+    c_out = r_out * (side_length_out / 2) / a_out
+    c_in = r_in * (side_length_in / 2) / a_in
 
-        # equivalent side length of half the corner radii triangular segment
-        c_out = r_out * (side_length_out / 2) / a_out
-        c_in = r_in * (side_length_in / 2) / a_in
+    # determine straight side length between corner radii (if present)
+    side_length_straight_out = side_length_out - (2 * c_out)
+    side_length_straight_in = side_length_in - (2 * c_in)
 
-        # determine straight side length between corner radii (if present)
-        side_length_straight_out = side_length_out - (2 * c_out)
-        side_length_straight_in = side_length_in - (2 * c_in)
+    # temp list for repeating geometry
+    outer_base_points = []
+    inner_base_points = []
 
-        # assign control point central on bottom side length & rotate to initial rotation specified
-        control_points = [self.rotate([0, -a_out + t / 2], rot)]
+    # start at bottom face, constructing one corner radii, then rotate by initial rotation +
+    # alpha and repeat for n_side number of times to form full section perimeter
 
-        super().__init__(control_points, shift)
+    # construct the first radius (bottom right)
+    for i in range(n_r):
+        # determine polar angle
+        theta = 1 / 2 * np.pi + i * 1.0 / max(1, n_r - 1) * alpha
 
-        # temp list for repeating geometry
-        base_points = []
+        # calculate location of inner and outer points
+        x_outer = side_length_straight_out / 2 - r_out * np.cos(theta)
+        y_outer = -a_out + r_out - r_out * np.sin(theta)
+        x_inner = side_length_straight_in / 2 - r_in * np.cos(theta)
+        y_inner = -a_in + r_in - r_in * np.sin(theta)
 
-        # specify a hole in the centre of the Polygon section
-        self.holes = [[0, 0]]
+        # append the current temporary points to the temporary points list
+        outer_base_points.append([x_outer, y_outer])
+        inner_base_points.append([x_inner, y_inner])
 
-        # start at bottom face, constructing one corner radii, then rotate by initial rotation +
-        # alpha and repeat for n_side number of times to form full section perimeter
+    # if radii merged to circle with an outer diameter of a_out then skip last point as causes
+    # overlapping end points which causes meshing issues if geometry is not cleaned by user
```

Comment:
This might no longer be required with shapely now handling the geometry rather than relying on prior clean_geometry action.

Have not tested polygon_section yet though.

Reply:
Removed!